definitions/procedures/pulp/remove.rb

```diff
@@ -0,0 +1,91 @@
+module Procedures::Pulp
+  class Remove < ForemanMaintain::Procedure
+    metadata do
+      description 'Remove pulp2'
+      for_feature :pulp2
+    end
+
+    def sys
+      ForemanMaintain::Utils::SystemHelpers.new
+    end
+
+    def pulp_data_dirs
+      [
+        feature(:pulp2).data_dir + '/published',
+        feature(:pulp2).data_dir + '/content',
+        feature(:pulp2).data_dir + '/importers',
+        feature(:pulp2).data_dir + '/uploads',
+        feature(:mongo).data_dir
+      ]
+    end
+
+    def pulp_packages
+      [
+        'pulp-server', 'python-pulp-streamer', 'pulp-puppet-plugins',
```

Comment:
Remember that this also removes `katello` since `katello` depends on `pulp-server`:
https://github.com/theforeman/foreman-packaging/blob/00695f781b79b926b9c6bbb71a30b8b543c51015/packages/katello/katello/katello.spec#L41-L59

Then `katello-common` depends on `pulp-selinux`:

https://github.com/theforeman/foreman-packaging/blob/00695f781b79b926b9c6bbb71a30b8b543c51015/packages/katello/katello/katello.spec#L122-L125

Reply:
i assume those deps will get removed once the katello-agent stuff goes in and we aren't installing pulp2 anymore. 

@jjeffers its probably a good idea to double check we're running katello 4.0.  I'd probably use check_min_version to test the katello-common package so it'll run on katello and external smart proxies